src/geometry/Shape.ts

```diff
@@ -343,4 +368,15 @@ export default abstract class Shape extends THREE.Group {
     }
     return target;
   }
-}
+
+  addLabel(tex: string, direction: THREE.Vector3) {
+    // Create a Text object for the label
+    const label = new Text(tex);
+
+    // Position the label using moveNextTo
+    moveNextTo(label, this, direction);
+
+    // Add the label as a child of the shape
+    this.add(label);
```

Comment:
Add the label first and position it afterward since adding it may change it's size/position in the scene.